File: VIQA/HtmlElements/SimpleElements/Button.cs
using OpenQA.Selenium;
using VIQA.HtmlElements.Interfaces;

namespace VIQA.HtmlElements
{
    public class Button : ClickableText, IButton
    {
        private const string LocatorTmpl = "input[type=button][{0}={1}]";
        public static string CommonLocatorById(string id) { return string.Format(LocatorTmpl, "id", id); }
        public static string CommonLocatorByNamed(string id) { return string.Format(LocatorTmpl, "name", id); }
        public static string CommonLocatorByClassName(string id) { return string.Format(LocatorTmpl, "class", id); }

        protected override string _typeName { get { return "Button"; } }

        public Button() {  }
        public Button(string name) : base(name) { Init(); }
        public Button(string name, string cssSelector) : base(name, cssSelector) { Init(); }
        public Button(string name, By byLocator) : base(name, byLocator) { Init(); }
        public Button(By byLocator) : base(byLocator) { Init(); }
        public Button(string name, IWebElement webElement) : base(name, webElement) { Init(); }
        public Button(IWebElement webElement) : base(webElement) { Init(); }

        private void Init()
        {
            TextElement.GetTextFunc = () => GetWebElement().GetAttribute("value");
        }

    }
}

File: Java/src/main/java/VIElements/SimpleElements/Button.java
package VIElements.SimpleElements;

import VIElements.Interfaces.*;
import org.openqa.selenium.*;

import static java.lang.String.format;

/**
 * Created by 12345 on 02.10.2014.
 */
public class Button extends ClickableText implements IButton {
    private static final String LocatorTmpl = "input[type=button][%s=%s]";
    public static String commonLocatorById(String id) { return format(LocatorTmpl, "id", id); }
    public static String commonLocatorByNamed(String id) { return format(LocatorTmpl, "name", id); }
    public static String commonLocatorByClassName(String id) { return format(LocatorTmpl, "class", id); }

    public Button() throws Exception{ super(); TypeName = "Button";}
    public Button(String name) throws Exception { super(name); }
    public Button(String name, String cssSelector) throws Exception { super(name, cssSelector); }
    public Button(String name, By byLocator) throws Exception { super(name, byLocator); }
    public Button(By byLocator) throws Exception { super(byLocator); }
    public Button(By byClickLocator, By byTextLocator) throws Exception { super(byClickLocator, byTextLocator); }
    public Button(String name, WebElement webElement) throws Exception { super(name, webElement); }
    public Button(WebElement webElement) throws Exception { super(webElement); }

    @Override
    public String getTextAction() throws Exception {
        return getUniqueWebElement().getAttribute("value");
    }
}
